www/nginx/src/opnsense/mvc/app/models/OPNsense/Nginx/Migrations/M1_24_0.php

```diff
@@ -29,25 +29,36 @@
 namespace OPNsense\Nginx\Migrations;
 
 use OPNsense\Base\BaseModelMigration;
+use OPNsense\Core\Config;
 
 class M1_24_0 extends BaseModelMigration
 {
     public function run($model)
     {
-        foreach ($model->getNodeByReference('http_server')->iterateItems() as $http_server) {
-            if ($http_server->listen_http_port != '') {
-                $http_server->listen_http_address = $http_server->listen_http_port . ',[::]:' . $http_server->listen_http_port;
-                $http_server->listen_http_port = null;
+        $cfgObj = Config::getInstance()->object();
+        $ports = array();
+        if (!empty($cfgObj->OPNsense->Nginx)) {
+            foreach ($cfgObj->OPNsense->Nginx->http_server as $cfg_http_server) {
+                $uuid = (string)$cfg_http_server->attributes()['uuid'];
+                $ports[$uuid]['http_port'] = (isset($cfg_http_server->listen_http_port) && $cfg_http_server->listen_http_port != '') ? $cfg_http_server->listen_http_port : null;
```

Comment:
Maybe you can write it directly to the model?

```suggestion
                $node = $mdl->getNodeByReference("http_server.$uuid");
                $port = (isset($cfg_http_server->listen_http_port) && $cfg_http_server->listen_http_port != '') ? $cfg_http_server->listen_http_port : null;
                $node->listen_http_address = (isset($port)) ? "$port,[::]:$port" : null;
```

Just an idea. Not tested...

Reply:
On the other hand, performance is not really an issue for model migrations... ;-)